lxd/endpoints/network.go

```diff
@@ -172,6 +174,31 @@ func (e *Endpoints) NetworkUpdateTrustedProxy(trustedProxy string) {
 	if ok && listener != nil {
 		listener.(*listeners.FancyTLSListener).TrustedProxy(proxies)
 	}
+
+	server, ok := e.servers[network]
+	if ok && server != nil {
+		server.ErrorLog = log.New(discard{proxies: proxies}, "", 0)
+	}
+}
+
+type discard struct {
+	proxies []net.IP
+}
+
+// Regex for the log we want to ignore.
+var unwantedLogRegex = regexp.MustCompile("http: TLS handshake error from (.*): .*")
```

Comment:
I think TLS handshakes can fail for various reasons so I'd suggest being really specific and look for ` write: connection reset by peer$` to detect when the underlying TCP connection is reset.

Reply:
yeah, we went through it with @markylaing in an earlier meeting, he's going to rework it to filter specifically that error and the haproxy servers.

Reply:
Thanks @simondeziel I've updated the regex and added some unit tests.

I haven't gone as far as testing this with haproxy itself, do you think that's required?